<?xml version="1.0"?>
<!--
NOTICE: this configuration is not a running example.
It contains only all services and their parameter for current package.
Some functional examples you can find in iop_cfg_sim_stage_fkie package!
-->
<launch>
  <arg name="ns" default="robot"/>
  <arg name="id_subsystem" default="127"/>
  <arg name="id_node" default="64"/>
  <group ns="$(arg ns)">
    <node name="iop_platform_manager" pkg="iop_component_fkie" type="iop_component">
        <param name="capability_group" value="iop_bridge"/>
        <param name="name_node" value="platform"/>
        <param name="iop_address" value="$(arg id_subsystem).$(arg id_node).49"/>
        <rosparam param="services">
            [
            iop_transport_fkie: "Transport",
            iop_events_fkie: "Events",
            iop_accesscontrol_fkie: "AccessControl",
            iop_discovery_fkie: "Discovery",
            iop_discovery_fkie: "DiscoveryClient",
            iop_management_fkie: "Management",
            iop_liveness_fkie: "Liveness",
            iop_manipulator_specification_service_fkie: "ManipulatorSpecificationService",
            iop_primitive_manipulator_fkie: "PrimitiveManipulator",
            iop_primitive_endeffector_fkie: "PrimitiveEndEffector",
            ]
        </rosparam>
        <rosparam subst_value="true">
          Events:
            events_timeout: 30
          AccessControl:
            access_timeout: 60
          Discovery:
            system_id: 2
            system_type: 10001
          DiscoveryClient:
            register_own_services: true
            enable_ros_interface: false
            force_component_update_after: 300
          ManipulatorSpecificationService:
            manipulator_base_link: manipulator_base_link
            manipulator_tip_link: gripperTurn_link
            robot_description: $(arg ns)/robot_description
          PrimitiveManipulator:
            NoParams: 0
          PrimitiveEndEffector:
            gripper_joint: gripperWidth_joint
            manipulator_id: $(arg id_subsystem).$(arg id_node).49
        </rosparam>
    </node>
  </group>
</launch>